teamcity: failed tests on master: testrace/TestAmbiguousCommit, testrace/TestAmbiguousCommit/ambiguousSuccess=false

The following tests appear to have failed:

[#307365](https://teamcity.cockroachdb.com/viewLog.html?buildId=307365):

```
--- FAIL: testrace/TestAmbiguousCommit (11.740s)
null
--- FAIL: testrace/TestAmbiguousCommit/ambiguousSuccess=false (5.810s)
ambiguous_commit_test.go:187: pq: result is ambiguous (error=TestAmbiguousCommit/ambiguousSuccess=false: RPC error: success=false)
ambiguous_commit_test.go:197: expected 1 row(s) but found 0
------- Stdout: -------
W170727 20:08:14.460980 30303 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170727 20:08:14.462216 30303 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170727 20:08:14.468412 30303 server/config.go:523  [n?] 1 storage engine initialized
I170727 20:08:14.468500 30303 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170727 20:08:14.468538 30303 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170727 20:08:14.502965 30303 server/node.go:336  [n?] **** cluster f204b79f-823b-44cd-8ca3-af02fb6eeefe has been created
I170727 20:08:14.508569 30303 server/server.go:781  [n?] **** add additional nodes by specifying --join=127.0.0.1:36638
I170727 20:08:14.535847 30303 storage/store.go:1262  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170727 20:08:14.536146 30303 server/node.go:421  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:0 WritesPerSecond:0}
I170727 20:08:14.536394 30303 server/node.go:320  [n1] node ID 1 initialized
I170727 20:08:14.536642 30303 gossip/gossip.go:297  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:36638" > attrs:<> locality:<> 
I170727 20:08:14.563824 30303 storage/stores.go:295  [n1] read 0 node addresses from persistent storage
I170727 20:08:14.565079 30303 server/node.go:556  [n1] connecting to gossip network to verify cluster ID...
I170727 20:08:14.565322 30303 server/node.go:581  [n1] node connected via gossip and verified as part of cluster "f204b79f-823b-44cd-8ca3-af02fb6eeefe"
I170727 20:08:14.565548 30384 storage/replica_command.go:2672  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I170727 20:08:14.565915 30303 server/node.go:362  [n1] node=1: started with [=] engine(s) and attributes []
I170727 20:08:14.569272 30303 sql/executor.go:419  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:36638}
I170727 20:08:14.664909 30303 server/server.go:860  [n1] starting https server at 127.0.0.1:44037
I170727 20:08:14.665043 30303 server/server.go:861  [n1] starting grpc/postgres server at 127.0.0.1:36638
I170727 20:08:14.665094 30303 server/server.go:862  [n1] advertising CockroachDB node at 127.0.0.1:36638
E170727 20:08:14.669662 30385 storage/queue.go:658  [replicate,n1,s1,r1/1:/{Min-System/}] range requires a replication change, but lacks a quorum of live replicas (0/1)
I170727 20:08:14.678243 30384 storage/replica_command.go:2672  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/tsd [r3]
E170727 20:08:14.736792 30385 storage/queue.go:658  [replicate,n1,s1,r1/1:/{Min-System/}] range requires a replication change, but lacks a quorum of live replicas (0/1)
E170727 20:08:14.783255 30610 storage/replica_proposal.go:539  [n1,s1,r2/1:/{System/-Max}] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
I170727 20:08:14.791668 30384 storage/replica_command.go:2672  [split,n1,s1,r3/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r4]
I170727 20:08:14.848315 30303 sql/event_log.go:101  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170727 20:08:14.942020 30384 storage/replica_command.go:2672  [split,n1,s1,r4/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r5]
I170727 20:08:14.992444 30384 storage/replica_command.go:2672  [split,n1,s1,r5/1:/{Table/System���-Max}] initiating a split of this range at key /Table/11 [r6]
I170727 20:08:15.028299 30303 sql/lease.go:367  [n1] publish: descID=12 (eventlog) version=2 mtime=2017-07-27 20:08:15.028190671 +0000 UTC
I170727 20:08:15.080988 30384 storage/replica_command.go:2672  [split,n1,s1,r6/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r7]
I170727 20:08:15.130081 30384 storage/replica_command.go:2672  [split,n1,s1,r7/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r8]
I170727 20:08:15.195439 30384 storage/replica_command.go:2672  [split,n1,s1,r8/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r9]
I170727 20:08:15.237206 30303 server/server.go:994  [n1] done ensuring all necessary migrations have run
I170727 20:08:15.237411 30303 server/server.go:996  [n1] serving sql connections
I170727 20:08:15.285273 30816 sql/event_log.go:101  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:36638} Attrs: Locality:} ClusterID:f204b79f-823b-44cd-8ca3-af02fb6eeefe StartedAt:1501186094565390990 LastUp:1501186094565390990}
I170727 20:08:15.309558 30384 storage/replica_command.go:2672  [split,n1,s1,r9/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r10]
W170727 20:08:15.488799 30303 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170727 20:08:15.490229 30303 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W170727 20:08:15.504679 30303 gossip/gossip.go:1196  [n?] no incoming or outgoing connections
I170727 20:08:15.535025 30303 server/config.go:523  [n?] 1 storage engine initialized
I170727 20:08:15.535176 30303 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170727 20:08:15.535212 30303 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170727 20:08:15.535382 30303 server/server.go:783  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I170727 20:08:15.620207 30958 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:36638
I170727 20:08:15.625019 31111 gossip/server.go:234  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:35637}
I170727 20:08:15.655542 30303 server/node.go:408  [n?] store [n0,s0] not bootstrapped
I170727 20:08:15.655705 30303 storage/stores.go:295  [n?] read 0 node addresses from persistent storage
I170727 20:08:15.655840 30303 server/node.go:556  [n?] connecting to gossip network to verify cluster ID...
I170727 20:08:15.656543 31021 storage/stores.go:314  [n?] wrote 1 node addresses to persistent storage
I170727 20:08:15.658146 30303 server/node.go:581  [n?] node connected via gossip and verified as part of cluster "f204b79f-823b-44cd-8ca3-af02fb6eeefe"
I170727 20:08:15.677809 30303 kv/dist_sender.go:364  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170727 20:08:15.688086 30303 server/node.go:313  [n?] new node allocated ID 2
I170727 20:08:15.688414 30303 gossip/gossip.go:297  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:35637" > attrs:<> locality:<> 
I170727 20:08:15.689300 30303 server/node.go:362  [n2] node=2: started with [=] engine(s) and attributes []
I170727 20:08:15.692022 30303 sql/executor.go:419  [n2] creating distSQLPlanner with address {tcp 127.0.0.1:35637}
I170727 20:08:15.699222 30986 storage/stores.go:314  [n1] wrote 1 node addresses to persistent storage
I170727 20:08:15.758070 30303 server/server.go:860  [n2] starting https server at 127.0.0.1:39653
I170727 20:08:15.758202 30303 server/server.go:861  [n2] starting grpc/postgres server at 127.0.0.1:35637
I170727 20:08:15.758239 30303 server/server.go:862  [n2] advertising CockroachDB node at 127.0.0.1:35637
I170727 20:08:15.763419 31104 server/node.go:537  [n2] bootstrapped store [n2,s2]
I170727 20:08:15.789892 30303 server/server.go:994  [n2] done ensuring all necessary migrations have run
I170727 20:08:15.790035 30303 server/server.go:996  [n2] serving sql connections
I170727 20:08:15.921745 31245 sql/event_log.go:101  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:35637} Attrs: Locality:} ClusterID:f204b79f-823b-44cd-8ca3-af02fb6eeefe StartedAt:1501186095688959987 LastUp:1501186095688959987}
W170727 20:08:15.960283 30303 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170727 20:08:15.961558 30303 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W170727 20:08:15.978429 30303 gossip/gossip.go:1196  [n?] no incoming or outgoing connections
I170727 20:08:15.983851 30303 server/config.go:523  [n?] 1 storage engine initialized
I170727 20:08:15.983953 30303 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170727 20:08:15.983985 30303 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170727 20:08:15.984181 30303 server/server.go:783  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I170727 20:08:16.159410 31119 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:36638
I170727 20:08:16.161383 31254 gossip/server.go:234  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:54813}
I170727 20:08:16.181245 30303 server/node.go:408  [n?] store [n0,s0] not bootstrapped
I170727 20:08:16.181759 30303 storage/stores.go:295  [n?] read 0 node addresses from persistent storage
I170727 20:08:16.183876 30303 server/node.go:556  [n?] connecting to gossip network to verify cluster ID...
I170727 20:08:16.183995 30303 server/node.go:581  [n?] node connected via gossip and verified as part of cluster "f204b79f-823b-44cd-8ca3-af02fb6eeefe"
I170727 20:08:16.189142 31353 storage/stores.go:314  [n?] wrote 1 node addresses to persistent storage
I170727 20:08:16.189855 31353 storage/stores.go:314  [n?] wrote 2 node addresses to persistent storage
I170727 20:08:16.202071 30303 kv/dist_sender.go:364  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170727 20:08:16.207671 30303 server/node.go:313  [n?] new node allocated ID 3
I170727 20:08:16.207995 30303 gossip/gossip.go:297  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:54813" > attrs:<> locality:<> 
I170727 20:08:16.209279 30303 server/node.go:362  [n3] node=3: started with [=] engine(s) and attributes []
I170727 20:08:16.211526 31378 storage/stores.go:314  [n1] wrote 2 node addresses to persistent storage
I170727 20:08:16.213031 30303 sql/executor.go:419  [n3] creating distSQLPlanner with address {tcp 127.0.0.1:54813}
I170727 20:08:16.227491 31379 storage/stores.go:314  [n2] wrote 2 node addresses to persistent storage
I170727 20:08:16.305964 30303 server/server.go:860  [n3] starting https server at 127.0.0.1:42700
I170727 20:08:16.306653 30303 server/server.go:861  [n3] starting grpc/postgres server at 127.0.0.1:54813
I170727 20:08:16.306710 30303 server/server.go:862  [n3] advertising CockroachDB node at 127.0.0.1:54813
I170727 20:08:16.315520 30303 server/server.go:994  [n3] done ensuring all necessary migrations have run
I170727 20:08:16.315647 30303 server/server.go:996  [n3] serving sql connections
I170727 20:08:16.335179 31259 server/node.go:537  [n3] bootstrapped store [n3,s3]
I170727 20:08:16.367956 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r10/1:/{Table/15-Max}] generated preemptive snapshot 9740a894 at index 11
I170727 20:08:16.394565 31274 sql/event_log.go:101  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:{NetworkField:tcp AddressField:127.0.0.1:54813} Attrs: Locality:} ClusterID:f204b79f-823b-44cd-8ca3-af02fb6eeefe StartedAt:1501186096208597078 LastUp:1501186096208597078}
I170727 20:08:16.479651 30687 storage/store.go:3477  [replicate,n1,s1,r10/1:/{Table/15-Max}] streamed snapshot to (n2,s2):?: kv pairs: 9, log entries: 1, rate-limit: 8.0 MiB/sec, 3ms
I170727 20:08:16.481086 31344 storage/replica_raftstorage.go:720  [n2,s2,r10/?:{-}] applying preemptive snapshot at index 11 (id=9740a894, encoded size=547, 1 rocksdb batches, 1 log entries)
I170727 20:08:16.482430 31344 storage/replica_raftstorage.go:728  [n2,s2,r10/?:/{Table/15-Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170727 20:08:16.503001 30687 storage/replica_command.go:3605  [replicate,n1,s1,r10/1:/{Table/15-Max}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r10:/{Table/15-Max} [(n1,s1):1, next=2]
I170727 20:08:16.523310 31536 storage/replica.go:2942  [n1,s1,r10/1:/{Table/15-Max}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I170727 20:08:16.528419 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r8/1:/Table/1{3-4}] generated preemptive snapshot 3355cae1 at index 25
I170727 20:08:16.543054 31605 storage/raft_transport.go:453  [n2] raft transport stream to node 1 established
I170727 20:08:16.697839 30687 storage/store.go:3477  [replicate,n1,s1,r8/1:/Table/1{3-4}] streamed snapshot to (n3,s3):?: kv pairs: 74, log entries: 15, rate-limit: 8.0 MiB/sec, 9ms
I170727 20:08:16.699953 31553 storage/replica_raftstorage.go:720  [n3,s3,r8/?:{-}] applying preemptive snapshot at index 25 (id=3355cae1, encoded size=21423, 1 rocksdb batches, 15 log entries)
I170727 20:08:16.702200 31553 storage/replica_raftstorage.go:728  [n3,s3,r8/?:/Table/1{3-4}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170727 20:08:16.705473 30687 storage/replica_command.go:3605  [replicate,n1,s1,r8/1:/Table/1{3-4}] change replicas (ADD_REPLICA (n3,s3):2): read existing descriptor r8:/Table/1{3-4} [(n1,s1):1, next=2]
I170727 20:08:16.729091 31625 storage/replica.go:2942  [n1,s1,r8/1:/Table/1{3-4}] proposing ADD_REPLICA((n3,s3):2): updated=[(n1,s1):1 (n3,s3):2] next=3
I170727 20:08:16.737744 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r7/1:/Table/1{2-3}] generated preemptive snapshot 0e7b9e93 at index 24
I170727 20:08:16.778933 30687 storage/store.go:3477  [replicate,n1,s1,r7/1:/Table/1{2-3}] streamed snapshot to (n2,s2):?: kv pairs: 31, log entries: 14, rate-limit: 8.0 MiB/sec, 33ms
I170727 20:08:16.780267 31609 storage/replica_raftstorage.go:720  [n2,s2,r7/?:{-}] applying preemptive snapshot at index 24 (id=0e7b9e93, encoded size=16267, 1 rocksdb batches, 14 log entries)
I170727 20:08:16.782013 31609 storage/replica_raftstorage.go:728  [n2,s2,r7/?:/Table/1{2-3}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170727 20:08:16.784260 31630 storage/raft_transport.go:453  [n3] raft transport stream to node 1 established
I170727 20:08:16.809846 30687 storage/replica_command.go:3605  [replicate,n1,s1,r7/1:/Table/1{2-3}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r7:/Table/1{2-3} [(n1,s1):1, next=2]
I170727 20:08:16.882610 31591 storage/replica.go:2942  [n1,s1,r7/1:/Table/1{2-3}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I170727 20:08:16.888584 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r4/1:/{System/tse-Table/System���}] generated preemptive snapshot 3a9a6d0a at index 22
I170727 20:08:16.893734 30687 storage/store.go:3477  [replicate,n1,s1,r4/1:/{System/tse-Table/System���}] streamed snapshot to (n3,s3):?: kv pairs: 12, log entries: 12, rate-limit: 8.0 MiB/sec, 3ms
I170727 20:08:16.895726 31457 storage/replica_raftstorage.go:720  [n3,s3,r4/?:{-}] applying preemptive snapshot at index 22 (id=3a9a6d0a, encoded size=9106, 1 rocksdb batches, 12 log entries)
I170727 20:08:16.897333 31457 storage/replica_raftstorage.go:728  [n3,s3,r4/?:/{System/tse-Table/System���}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170727 20:08:16.903646 30687 storage/replica_command.go:3605  [replicate,n1,s1,r4/1:/{System/tse-Table/System���}] change replicas (ADD_REPLICA (n3,s3):2): read existing descriptor r4:/{System/tse-Table/SystemConfigSpan/Start} [(n1,s1):1, next=2]
I170727 20:08:16.954890 31698 storage/replica.go:2942  [n1,s1,r4/1:/{System/tse-Table/System���}] proposing ADD_REPLICA((n3,s3):2): updated=[(n1,s1):1 (n3,s3):2] next=3
I170727 20:08:16.997836 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r2/1:/System/{-tsd}] generated preemptive snapshot 8e207318 at index 41
I170727 20:08:17.069726 30687 storage/store.go:3477  [replicate,n1,s1,r2/1:/System/{-tsd}] streamed snapshot to (n2,s2):?: kv pairs: 32, log entries: 3, rate-limit: 8.0 MiB/sec, 63ms
I170727 20:08:17.073417 31671 storage/replica_raftstorage.go:720  [n2,s2,r2/?:{-}] applying preemptive snapshot at index 41 (id=8e207318, encoded size=74359, 1 rocksdb batches, 3 log entries)
I170727 20:08:17.075836 31671 storage/replica_raftstorage.go:728  [n2,s2,r2/?:/System/{-tsd}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170727 20:08:17.080745 30687 storage/replica_command.go:3605  [replicate,n1,s1,r2/1:/System/{-tsd}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r2:/System/{-tsd} [(n1,s1):1, next=2]
I170727 20:08:17.113025 31730 storage/replica.go:2942  [n1,s1,r2/1:/System/{-tsd}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I170727 20:08:17.119044 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r1/1:/{Min-System/}] generated preemptive snapshot 5d583c5f at index 51
I170727 20:08:17.125797 30687 storage/store.go:3477  [replicate,n1,s1,r1/1:/{Min-System/}] streamed snapshot to (n3,s3):?: kv pairs: 37, log entries: 41, rate-limit: 8.0 MiB/sec, 6ms
I170727 20:08:17.128372 31681 storage/replica_raftstorage.go:720  [n3,s3,r1/?:{-}] applying preemptive snapshot at index 51 (id=5d583c5f, encoded size=24006, 1 rocksdb batches, 41 log entries)
I170727 20:08:17.130985 31681 storage/replica_raftstorage.go:728  [n3,s3,r1/?:/{Min-System/}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170727 20:08:17.148409 30687 storage/replica_command.go:3605  [replicate,n1,s1,r1/1:/{Min-System/}] change replicas (ADD_REPLICA (n3,s3):2): read existing descriptor r1:/{Min-System/} [(n1,s1):1, next=2]
I170727 20:08:17.183052 31720 storage/replica.go:2942  [n1,s1,r1/1:/{Min-System/}] proposing ADD_REPLICA((n3,s3):2): updated=[(n1,s1):1 (n3,s3):2] next=3
I170727 20:08:17.195972 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r6/1:/Table/1{1-2}] generated preemptive snapshot 6ae09c5a at index 18
I170727 20:08:17.216880 30687 storage/store.go:3477  [replicate,n1,s1,r6/1:/Table/1{1-2}] streamed snapshot to (n2,s2):?: kv pairs: 11, log entries: 8, rate-limit: 8.0 MiB/sec, 5ms
I170727 20:08:17.217038 31739 storage/replica_raftstorage.go:720  [n2,s2,r6/?:{-}] applying preemptive snapshot at index 18 (id=6ae09c5a, encoded size=6398, 1 rocksdb batches, 8 log entries)
I170727 20:08:17.219586 31739 storage/replica_raftstorage.go:728  [n2,s2,r6/?:/Table/1{1-2}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I170727 20:08:17.223868 30687 storage/replica_command.go:3605  [replicate,n1,s1,r6/1:/Table/1{1-2}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r6:/Table/1{1-2} [(n1,s1):1, next=2]
I170727 20:08:17.291866 31686 storage/replica.go:2942  [n1,s1,r6/1:/Table/1{1-2}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I170727 20:08:17.295897 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r5/1:/Table/{SystemCon���-11}] generated preemptive snapshot cbbb7e6d at index 27
I170727 20:08:17.322304 30687 storage/store.go:3477  [replicate,n1,s1,r5/1:/Table/{SystemCon���-11}] streamed snapshot to (n3,s3):?: kv pairs: 40, log entries: 17, rate-limit: 8.0 MiB/sec, 22ms
I170727 20:08:17.324535 31688 storage/replica_raftstorage.go:720  [n3,s3,r5/?:{-}] applying preemptive snapshot at index 27 (id=cbbb7e6d, encoded size=19424, 1 rocksdb batches, 17 log entries)
I170727 20:08:17.326611 31688 storage/replica_raftstorage.go:728  [n3,s3,r5/?:/Table/{SystemCon���-11}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170727 20:08:17.330431 30687 storage/replica_command.go:3605  [replicate,n1,s1,r5/1:/Table/{SystemCon���-11}] change replicas (ADD_REPLICA (n3,s3):2): read existing descriptor r5:/Table/{SystemConfigSpan/Start-11} [(n1,s1):1, next=2]
I170727 20:08:17.374494 31726 storage/replica.go:2942  [n1,s1,r5/1:/Table/{SystemCon���-11}] proposing ADD_REPLICA((n3,s3):2): updated=[(n1,s1):1 (n3,s3):2] next=3
I170727 20:08:17.381440 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r3/1:/System/ts{d-e}] generated preemptive snapshot f16c5275 at index 26
I170727 20:08:17.408751 30687 storage/store.go:3477  [replicate,n1,s1,r3/1:/System/ts{d-e}] streamed snapshot to (n2,s2):?: kv pairs: 908, log entries: 3, rate-limit: 8.0 MiB/sec, 26ms
I170727 20:08:17.411735 31729 storage/replica_raftstorage.go:720  [n2,s2,r3/?:{-}] applying preemptive snapshot at index 26 (id=f16c5275, encoded size=148490, 1 rocksdb batches, 3 log entries)
I170727 20:08:17.415502 31729 storage/replica_raftstorage.go:728  [n2,s2,r3/?:/System/ts{d-e}] applied preemptive snapshot in 3ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I170727 20:08:17.422754 30687 storage/replica_command.go:3605  [replicate,n1,s1,r3/1:/System/ts{d-e}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r3:/System/ts{d-e} [(n1,s1):1, next=2]
I170727 20:08:17.479847 31826 storage/replica.go:2942  [n1,s1,r3/1:/System/ts{d-e}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I170727 20:08:17.503164 30687 storage/replica_raftstorage.go:511  [replicate,n1,s1,r9/1:/Table/1{4-5}] generated preemptive snapshot bc64d5f1 at index 19
I170727 20:08:17.516348 30687 storage/store.go:3477  [replicate,n1,s1,r9/1:/Table/1{4-5}] streamed snapshot to (n3,s3):?: kv pairs: 10, log entries: 9, rate-limit: 8.0 MiB/sec, 3ms
I170727 20:08:17.519876 31814 storage/replica_raftstorage.go:720  [n3,s3,r9/?:{-}] applying preemptive snapshot at index 19 (id=bc64d5f1, encoded size=5868, 1 rocksdb batches, 9 log entries)
I170727 20:08:17.521328 31814 storage/replica_raftstorage.go:728  [n3,s3,r9/?:/Table/1{4-5}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170727 20:08:17.528025 30687 storage/replica_command.go:3605  [replicate,n1,s1,r9/1:/Table/1{4-5}] change replicas (ADD_REPLICA (n3,s3):2): read existing descriptor r9:/Table/1{4-5} [(n1,s1):1, next=2]
I170727 20:08:17.568499 31808 storage/replica.go:2942  [n1,s1,r9/1:/Table/1{4-5}] proposing ADD_REPLICA((n3,s3):2): updated=[(n1,s1):1 (n3,s3):2] next=3
I170727 20:08:17.574977 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r10/1:/{Table/15-Max}] generated preemptive snapshot c272d989 at index 16
I170727 20:08:17.578817 30385 storage/store.go:3477  [replicate,n1,s1,r10/1:/{Table/15-Max}] streamed snapshot to (n3,s3):?: kv pairs: 11, log entries: 6, rate-limit: 8.0 MiB/sec, 3ms
I170727 20:08:17.579759 30687 storage/queue.go:725  [n1,replicate] purgatory is now empty
I170727 20:08:17.593343 31565 storage/replica_raftstorage.go:720  [n3,s3,r10/?:{-}] applying preemptive snapshot at index 16 (id=c272d989, encoded size=3541, 1 rocksdb batches, 6 log entries)
I170727 20:08:17.597555 31565 storage/replica_raftstorage.go:728  [n3,s3,r10/?:/{Table/15-Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170727 20:08:17.610941 30385 storage/replica_command.go:3605  [replicate,n1,s1,r10/1:/{Table/15-Max}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r10:/{Table/15-Max} [(n1,s1):1, (n2,s2):2, next=3]
I170727 20:08:17.679632 31712 storage/replica.go:2942  [n1,s1,r10/1:/{Table/15-Max}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I170727 20:08:17.708268 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r7/1:/Table/1{2-3}] generated preemptive snapshot 82bd5a09 at index 27
I170727 20:08:17.740500 30385 storage/store.go:3477  [replicate,n1,s1,r7/1:/Table/1{2-3}] streamed snapshot to (n3,s3):?: kv pairs: 32, log entries: 17, rate-limit: 8.0 MiB/sec, 15ms
I170727 20:08:17.742702 31584 storage/replica_raftstorage.go:720  [n3,s3,r7/?:{-}] applying preemptive snapshot at index 27 (id=82bd5a09, encoded size=18298, 1 rocksdb batches, 17 log entries)
I170727 20:08:17.745760 31584 storage/replica_raftstorage.go:728  [n3,s3,r7/?:/Table/1{2-3}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170727 20:08:17.753894 30385 storage/replica_command.go:3605  [replicate,n1,s1,r7/1:/Table/1{2-3}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r7:/Table/1{2-3} [(n1,s1):1, (n2,s2):2, next=3]
I170727 20:08:17.791914 31859 storage/replica.go:2942  [n1,s1,r7/1:/Table/1{2-3}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I170727 20:08:17.803240 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r6/1:/Table/1{1-2}] generated preemptive snapshot f78917ff at index 21
I170727 20:08:17.811767 30385 storage/store.go:3477  [replicate,n1,s1,r6/1:/Table/1{1-2}] streamed snapshot to (n3,s3):?: kv pairs: 12, log entries: 11, rate-limit: 8.0 MiB/sec, 7ms
I170727 20:08:17.813498 31922 storage/replica_raftstorage.go:720  [n3,s3,r6/?:{-}] applying preemptive snapshot at index 21 (id=f78917ff, encoded size=8429, 1 rocksdb batches, 11 log entries)
I170727 20:08:17.836388 31922 storage/replica_raftstorage.go:728  [n3,s3,r6/?:/Table/1{1-2}] applied preemptive snapshot in 23ms [clear=0ms batch=0ms entries=0ms commit=17ms]
I170727 20:08:17.840254 30385 storage/replica_command.go:3605  [replicate,n1,s1,r6/1:/Table/1{1-2}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r6:/Table/1{1-2} [(n1,s1):1, (n2,s2):2, next=3]
I170727 20:08:17.906024 31863 storage/replica.go:2942  [n1,s1,r6/1:/Table/1{1-2}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I170727 20:08:17.922850 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r4/1:/{System/tse-Table/System���}] generated preemptive snapshot 38f45c30 at index 25
I170727 20:08:17.950813 30385 storage/store.go:3477  [replicate,n1,s1,r4/1:/{System/tse-Table/System���}] streamed snapshot to (n2,s2):?: kv pairs: 13, log entries: 15, rate-limit: 8.0 MiB/sec, 27ms
I170727 20:08:17.972599 31924 storage/replica_raftstorage.go:720  [n2,s2,r4/?:{-}] applying preemptive snapshot at index 25 (id=38f45c30, encoded size=11194, 1 rocksdb batches, 15 log entries)
I170727 20:08:17.975122 31924 storage/replica_raftstorage.go:728  [n2,s2,r4/?:/{System/tse-Table/System���}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I170727 20:08:17.984026 30385 storage/replica_command.go:3605  [replicate,n1,s1,r4/1:/{System/tse-Table/System���}] change replicas (ADD_REPLICA (n2,s2):3): read existing descriptor r4:/{System/tse-Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):2, next=3]
I170727 20:08:18.032351 31903 storage/replica.go:2942  [n1,s1,r4/1:/{System/tse-Table/System���}] proposing ADD_REPLICA((n2,s2):3): updated=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I170727 20:08:18.041783 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r1/1:/{Min-System/}] generated preemptive snapshot 98b03b04 at index 70
I170727 20:08:18.072495 30385 storage/store.go:3477  [replicate,n1,s1,r1/1:/{Min-System/}] streamed snapshot to (n2,s2):?: kv pairs: 48, log entries: 60, rate-limit: 8.0 MiB/sec, 30ms
I170727 20:08:18.074906 31877 storage/replica_raftstorage.go:720  [n2,s2,r1/?:{-}] applying preemptive snapshot at index 70 (id=98b03b04, encoded size=34261, 1 rocksdb batches, 60 log entries)
I170727 20:08:18.095015 31877 storage/replica_raftstorage.go:728  [n2,s2,r1/?:/{Min-System/}] applied preemptive snapshot in 9ms [clear=0ms batch=0ms entries=3ms commit=5ms]
I170727 20:08:18.099858 30385 storage/replica_command.go:3605  [replicate,n1,s1,r1/1:/{Min-System/}] change replicas (ADD_REPLICA (n2,s2):3): read existing descriptor r1:/{Min-System/} [(n1,s1):1, (n3,s3):2, next=3]
I170727 20:08:18.199773 31864 storage/replica.go:2942  [n1,s1,r1/1:/{Min-System/}] proposing ADD_REPLICA((n2,s2):3): updated=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I170727 20:08:18.239688 31474 storage/replica_proposal.go:454  [n3,s3,r10/3:/{Table/15-Max}] new range lease repl=(n3,s3):3 start=1501186098.210705430,0 epo=1 pro=1501186098.210713230,0 following repl=(n1,s1):1 start=0.000000000,0 exp=1501186103.532375684,0 pro=1501186094.532442484,0
I170727 20:08:18.248803 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r9/1:/Table/1{4-5}] generated preemptive snapshot 56ec5d82 at index 24
I170727 20:08:18.266941 30385 storage/store.go:3477  [replicate,n1,s1,r9/1:/Table/1{4-5}] streamed snapshot to (n2,s2):?: kv pairs: 12, log entries: 14, rate-limit: 8.0 MiB/sec, 16ms
I170727 20:08:18.268621 31916 storage/replica_raftstorage.go:720  [n2,s2,r9/?:{-}] applying preemptive snapshot at index 24 (id=56ec5d82, encoded size=8856, 1 rocksdb batches, 14 log entries)
I170727 20:08:18.270943 31916 storage/replica_raftstorage.go:728  [n2,s2,r9/?:/Table/1{4-5}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I170727 20:08:18.277199 30385 storage/replica_command.go:3605  [replicate,n1,s1,r9/1:/Table/1{4-5}] change replicas (ADD_REPLICA (n2,s2):3): read existing descriptor r9:/Table/1{4-5} [(n1,s1):1, (n3,s3):2, next=3]
I170727 20:08:18.375067 31966 storage/replica.go:2942  [n1,s1,r9/1:/Table/1{4-5}] proposing ADD_REPLICA((n2,s2):3): updated=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I170727 20:08:18.389976 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r3/1:/System/ts{d-e}] generated preemptive snapshot a6ba83a7 at index 29
I170727 20:08:18.449835 30385 storage/store.go:3477  [replicate,n1,s1,r3/1:/System/ts{d-e}] streamed snapshot to (n3,s3):?: kv pairs: 909, log entries: 6, rate-limit: 8.0 MiB/sec, 59ms
I170727 20:08:18.451891 31991 storage/replica_raftstorage.go:720  [n3,s3,r3/?:{-}] applying preemptive snapshot at index 29 (id=a6ba83a7, encoded size=150593, 1 rocksdb batches, 6 log entries)
I170727 20:08:18.465002 31991 storage/replica_raftstorage.go:728  [n3,s3,r3/?:/System/ts{d-e}] applied preemptive snapshot in 13ms [clear=0ms batch=0ms entries=1ms commit=9ms]
I170727 20:08:18.473674 30385 storage/replica_command.go:3605  [replicate,n1,s1,r3/1:/System/ts{d-e}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r3:/System/ts{d-e} [(n1,s1):1, (n2,s2):2, next=3]
I170727 20:08:18.477513 31986 storage/raft_transport.go:453  [n3] raft transport stream to node 2 established
I170727 20:08:18.552051 31889 storage/replica.go:2942  [n1,s1,r3/1:/System/ts{d-e}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I170727 20:08:18.566998 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r5/1:/Table/{SystemCon���-11}] generated preemptive snapshot 32134e87 at index 30
I170727 20:08:18.634505 30385 storage/store.go:3477  [replicate,n1,s1,r5/1:/Table/{SystemCon���-11}] streamed snapshot to (n2,s2):?: kv pairs: 41, log entries: 20, rate-limit: 8.0 MiB/sec, 67ms
I170727 20:08:18.645736 32083 storage/replica_raftstorage.go:720  [n2,s2,r5/?:{-}] applying preemptive snapshot at index 30 (id=32134e87, encoded size=21455, 1 rocksdb batches, 20 log entries)
I170727 20:08:18.655754 32083 storage/replica_raftstorage.go:728  [n2,s2,r5/?:/Table/{SystemCon���-11}] applied preemptive snapshot in 9ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170727 20:08:18.659932 30385 storage/replica_command.go:3605  [replicate,n1,s1,r5/1:/Table/{SystemCon���-11}] change replicas (ADD_REPLICA (n2,s2):3): read existing descriptor r5:/Table/{SystemConfigSpan/Start-11} [(n1,s1):1, (n3,s3):2, next=3]
I170727 20:08:18.715953 32092 storage/replica.go:2942  [n1,s1,r5/1:/Table/{SystemCon���-11}] proposing ADD_REPLICA((n2,s2):3): updated=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I170727 20:08:18.729217 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r2/1:/System/{-tsd}] generated preemptive snapshot 11109dc4 at index 44
I170727 20:08:18.733833 30385 storage/store.go:3477  [replicate,n1,s1,r2/1:/System/{-tsd}] streamed snapshot to (n3,s3):?: kv pairs: 33, log entries: 6, rate-limit: 8.0 MiB/sec, 4ms
I170727 20:08:18.736178 31975 storage/replica_raftstorage.go:720  [n3,s3,r2/?:{-}] applying preemptive snapshot at index 44 (id=11109dc4, encoded size=76402, 1 rocksdb batches, 6 log entries)
I170727 20:08:18.740146 31975 storage/replica_raftstorage.go:728  [n3,s3,r2/?:/System/{-tsd}] applied preemptive snapshot in 4ms [clear=0ms batch=0ms entries=1ms commit=2ms]
I170727 20:08:18.747350 30385 storage/replica_command.go:3605  [replicate,n1,s1,r2/1:/System/{-tsd}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r2:/System/{-tsd} [(n1,s1):1, (n2,s2):2, next=3]
I170727 20:08:18.766997 31888 storage/raft_transport.go:453  [n2] raft transport stream to node 3 established
I170727 20:08:18.809789 32073 storage/replica.go:2942  [n1,s1,r2/1:/System/{-tsd}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I170727 20:08:18.835387 30385 storage/replica_raftstorage.go:511  [replicate,n1,s1,r8/1:/Table/1{3-4}] generated preemptive snapshot e315fb5f at index 65
I170727 20:08:18.845321 30385 storage/store.go:3477  [replicate,n1,s1,r8/1:/Table/1{3-4}] streamed snapshot to (n2,s2):?: kv pairs: 166, log entries: 55, rate-limit: 8.0 MiB/sec, 9ms
I170727 20:08:18.854397 32112 storage/replica_raftstorage.go:720  [n2,s2,r8/?:{-}] applying preemptive snapshot at index 65 (id=e315fb5f, encoded size=69996, 1 rocksdb batches, 55 log entries)
I170727 20:08:18.878216 32112 storage/replica_raftstorage.go:728  [n2,s2,r8/?:/Table/1{3-4}] applied preemptive snapshot in 24ms [clear=0ms batch=0ms entries=18ms commit=4ms]
I170727 20:08:18.901348 30385 storage/replica_command.go:3605  [replicate,n1,s1,r8/1:/Table/1{3-4}] change replicas (ADD_REPLICA (n2,s2):3): read existing descriptor r8:/Table/1{3-4} [(n1,s1):1, (n3,s3):2, next=3]
I170727 20:08:18.985827 32075 storage/replica.go:2942  [n1,s1,r8/1:/Table/1{3-4}] proposing ADD_REPLICA((n2,s2):3): updated=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I170727 20:08:19.407698 32095 sql/event_log.go:101  [client=127.0.0.1:47065,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:test Statement:CREATE DATABASE test User:root}
I170727 20:08:19.453590 31495 storage/replica_command.go:2672  [split,n3,s3,r10/3:/{Table/15-Max}] initiating a split of this range at key /Table/50 [r21]
I170727 20:08:19.546431 32095 sql/event_log.go:101  [client=127.0.0.1:47065,user=root,n1] Event: "create_table", target: 51, info: {TableName:test.t Statement:CREATE TABLE test.t (k SERIAL PRIMARY KEY, v INT) User:root}
I170727 20:08:19.688678 31427 storage/replica_proposal.go:454  [n3,s3,r10/3:/{Table/15-Max}] new range lease repl=(n3,s3):3 start=1501186098.210705430,0 epo=1 pro=1501186098.210713230,0 following repl=(n3,s3):3 start=1501186098.210705430,0 epo=1 pro=1501186098.210713230,0
I170727 20:08:19.704364 31495 storage/replica_command.go:2672  [split,n3,s3,r21/3:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I170727 20:08:19.989223 31480 storage/replica_proposal.go:454  [n3,s3,r21/3:/{Table/50-Max}] new range lease repl=(n3,s3):3 start=1501186098.210705430,0 epo=1 pro=1501186098.210713230,0 following repl=(n3,s3):3 start=1501186098.210705430,0 epo=1 pro=1501186098.210713230,0
I170727 20:08:20.149799 32406 util/stop/stopper.go:473  quiescing; tasks left:
1      [async] drain
W170727 20:08:20.156021 31605 storage/raft_transport.go:459  [n2] raft transport stream to node 1 failed: rpc error: code = Unavailable desc = transport is closing
W170727 20:08:20.162663 31637 storage/raft_transport.go:459  [n1] raft transport stream to node 3 failed: rpc error: code = Unavailable desc = transport is closing
W170727 20:08:20.163054 31986 storage/raft_transport.go:459  [n3] raft transport stream to node 2 failed: EOF
W170727 20:08:20.163421 31630 storage/raft_transport.go:459  [n3] raft transport stream to node 1 failed: EOF
I170727 20:08:20.164675 30691 kv/transport_race.go:71  transport race promotion: ran 102 iterations on up to 677 requests
W170727 20:08:20.165014 31888 storage/raft_transport.go:459  [n2] raft transport stream to node 3 failed: rpc error: code = Unavailable desc = transport is closing
W170727 20:08:20.167674 31393 storage/raft_transport.go:459  [n1] raft transport stream to node 2 failed: rpc error: code = Unavailable desc = transport is closing

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/ambiguous_commit_test.go
+++ b/pkg/sql/ambiguous_commit_test.go
@@ -29,6 +29,8 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/rpc"
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -121,18 +123,25 @@ func TestAmbiguousCommit(t *testing.T) {
 				},
 			}
 
-			{
-				storeTestingKnobs := new(storage.StoreTestingKnobs)
-
-				if ambiguousSuccess {
-					storeTestingKnobs.TestingResponseFilter = func(args roachpb.BatchRequest, _ *roachpb.BatchResponse) *roachpb.Error {
+			if ambiguousSuccess {
+				params.Knobs.Store = &storage.StoreTestingKnobs{
+					TestingResponseFilter: func(args roachpb.BatchRequest, _ *roachpb.BatchResponse) *roachpb.Error {
 						if req, ok := args.GetArg(roachpb.ConditionalPut); ok {
 							return maybeRPCError(req.(*roachpb.ConditionalPutRequest))
 						}
 						return nil
-					}
+					},
 				}
-				params.Knobs.Store = storeTestingKnobs
+			}
+
+			// Avoid distSQL so we can reliably hydrate the intended dist
+			// sender's cache below.
+			{
+				distSQLOverride := &settings.EnumSetting{}
+				settings.TestingSetEnum(&distSQLOverride, int64(sql.DistSQLOff))
+				params.Knobs.SQLExecutor = &sql.ExecutorTestingKnobs{
+					OverrideDistSQLMode: distSQLOverride,
+				}
 			}
 
 			testClusterArgs := base.TestClusterArgs{

